log4j-core and log4j-api version constraint specified in io.quarkus:quarkus-bom does not match

### Describe the bug

Since version 3.0.1.Final, the io.quarkus:quarkus-bom package introduced a constraint on log4j-core and specified the version as 2.17.1. However, the version constraint for log4j-api remains at 2.20.0. It is unclear whether this was intentional, but it has led to a tricky bug during development (at least in my case, it was not easy to identify the root cause). You can find a reproducible repository at https://github.com/cuichenli/quarkus-log4j-core-constraint-issue.

When you run the program in this repo, you will see error:
```
Caused by: java.lang.NoSuchMethodError: 'java.lang.ClassLoader[] org.apache.logging.log4j.util.LoaderUtil.getClassLoaders()'
```

I have observed that when using io.quarkus.platform:quarkus-bom, this issue does not occur. Therefore, I'm curious about the scenarios in which we should use either the quarkus-bom package or the quarkus-platform:quarkus-bom package. Could you provide some guidance on this matter? Thank you!

--- a/bom/application/pom.xml
+++ b/bom/application/pom.xml
@@ -1894,13 +1894,6 @@
             </dependency>
             <dependency>
                 <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-reactive-datasource-deployment</artifactId>
-                <version>${project.version}</version>
-                <scope>test</scope>
-                <type>test-jar</type>
-            </dependency>
-            <dependency>
-                <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-reactive-db2-client</artifactId>
                 <version>${project.version}</version>
             </dependency>
@@ -2375,13 +2368,6 @@
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
                 <version>${project.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
-                <version>${project.version}</version>
-                <scope>test</scope>
-                <type>test-jar</type>
             </dependency>
             <dependency>
                 <groupId>io.quarkus</groupId>
@@ -6759,6 +6745,7 @@
                             <goal>flatten-platform-bom</goal>
                         </goals>
                         <configuration>
+                            <excludeScopes>test</excludeScopes>
                             <excludeArtifactKeys>
                                 <key>junit:junit</key> <!-- comes from the jackson-bom -->
                             </excludeArtifactKeys>

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -234,6 +234,20 @@
             </dependency>
             <dependency>
                 <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-reactive-datasource-deployment</artifactId>
+                <version>${project.version}</version>
+                <scope>test</scope>
+                <type>test-jar</type>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
+                <artifactId>quarkus-resteasy-reactive-deployment</artifactId>
+                <version>${project.version}</version>
+                <scope>test</scope>
+                <type>test-jar</type>
+            </dependency>
+            <dependency>
+                <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-war-launcher-runner</artifactId>
                 <version>${project.version}</version>
             </dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -61,7 +61,7 @@
 
         <nexus-staging-maven-plugin.version>1.6.8</nexus-staging-maven-plugin.version>
         <gitflow-incremental-builder.version>4.5.0</gitflow-incremental-builder.version>
-        <quarkus-platform-bom-plugin.version>0.0.92</quarkus-platform-bom-plugin.version>
+        <quarkus-platform-bom-plugin.version>0.0.93</quarkus-platform-bom-plugin.version>
 
         <skipDocs>false</skipDocs>
         <skip.gradle.tests>false</skip.gradle.tests>

